### DISCUSSION TITLE
Support for non-EVM compatible chains (e.g. Starknet)

### DISCUSSION THREAD

**Discussion by microHoffman** [wevm/viem#2477] [Idea / Feature Request]
Hello! 
Did you ever consider adding a support for non-EVM chains to Viem (and Wagmi), for example Starknet? I assume it would be quite a lot of work as there are many differences, but I was wondering if enabling something like this (e.g. via some adapters, interfaces or whatever) would be desirable or you want to stick strictly to EVM chains support?

The benefit of having this support would be that if a certain dapp supports both EVM & non-EVM chain (e.g. Starknet), the app can just stick to use viem/wagmi ecosystem and would not have to implement similar functionality related to all wallet/contract/signing/etc in some other package (e.g. starknet-js).

**Comment by tmm** [comment]
Unlikely our tools would support non-EVM chains. Think you are much better using Viem/Wagmi alongside whatever the best non-EVM tools are at the application-level versus us cramming everything into a Viem/Wagmi (since it would probably end with trade-offs making EVM experience worse).
